python/cudf/cudf/utils/ioutils.py

```diff
@@ -830,16 +841,24 @@ def get_filepath_or_buffer(path_or_data, compression, iotypes=(BytesIO)):
     compression : str
         Type of compression algorithm for the content
     """
-    if is_url(path_or_data):
-        with urllib.request.urlopen(path_or_data) as url:
-            compression = url.headers.get("Content-Encoding", None)
-            buffer = BytesIO(url.read())
-        return buffer, compression
-    elif isinstance(path_or_data, str):
-        return os.path.expanduser(path_or_data), compression
+    if isinstance(path_or_data, str):
+        storage_options = kwargs.get("storage_options")
+        # fsspec does not expanduser so handle here
+        path_or_data = os.path.expanduser(path_or_data)
+        fs, _, paths = fsspec.get_fs_token_paths(
+            path_or_data, mode="rb", storage_options=storage_options
+        )
+        if _is_local_filesystem(fs):
+            # Doing this as `read_json` accepts a json string
+            # path_or_data need not be a filepath like string
+            if os.path.exists(paths[0]):
```

Comment:
Doing this because in the case of `read_json` path_or_data is allowed to be a python `str` and does not need to be a buffer or I/O like object.